### PR TITLE
Update corrade recipe

### PR DESCRIPTION
**Pull request by danimtb**
Specify library name and version:  **corrade/all**

closes https://github.com/conan-io/conan-center-index/issues/1397

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/corrade/all/conanfile.py [hunk 1]
```diff
@@ -65,6 +48,7 @@ def configure(self):
         if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version.value) < 14:
             raise ConanInvalidConfiguration("Corrade requires Visual Studio version 14 or greater")
         if tools.cross_building(self.settings):
```
Comment (jgsogo) [inline recipes/corrade/all/conanfile.py:50]:
```suggestion
        if tools.cross_building(self):
```

File: recipes/corrade/all/conanfile.py [hunk 2]
```diff
@@ -65,6 +48,7 @@ def configure(self):
         if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version.value) < 14:
```
Comment (jgsogo) [inline recipes/corrade/all/conanfile.py:48]:
```suggestion
        if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version) < 14:
```

**Comment by danimtb** [comment]
So the problem here was clearly on the `tools.cross_building()` and the `ConanInvalidConfiguration`.

As part of the pipeline, we iterate the recipe with `conan info` for all the profiles we are using and checking if the package ID is the same for some of them in order to discard any build that is not needed. However, this process is done in a Linux machine and it detects that using a Windows or Linux profile, you are cross-building. FInally `ConanInvalidConfiguration` prevents the pipeline from raising, as it is considered an expected error.

I have removed the exception there and left a warning just in case.

File: recipes/corrade/all/conanfile.py (hunk 1 above)
Comment (Croydon) [reply recipes/corrade/all/conanfile.py]: This is the recommend way now? Good to know :smile:

Comment (danimtb) [reply recipes/corrade/all/conanfile.py]: yes, this was done in order to support the new cross-building model

**Comment by Croydon** [comment]
> So the problem here was clearly on the `tools.cross_building()` and the `ConanInvalidConfiguration`.
> 
> As part of the pipeline, we iterate the recipe with `conan info` for all the profiles we are using and checking if the package ID is the same for some of them in order to discard any build that is not needed. However, this process is done in a Linux machine and it detects that using a Windows or Linux profile, you are cross-building. FInally `ConanInvalidConfiguration` prevents the pipeline from raising, as it is considered an expected error.
> 
> I have removed the exception there and left a warning just in case.

Thanks for figuring this one out! 

Sounds like something which could happen more often as this is an implementation detail which isn't really in the mind of recipe authors :thinking:

File: recipes/corrade/all/conanfile.py [hunk 3]
```diff
@@ -62,10 +45,10 @@ def config_options(self):
             del self.options.fPIC
 
     def configure(self):
-        if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version.value) < 14:
+        if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version) < 14:
             raise ConanInvalidConfiguration("Corrade requires Visual Studio version 14 or greater")
-        if tools.cross_building(self.settings):
-            raise ConanInvalidConfiguration("This Corrade recipe doesn't support cross building yet")
+        if tools.cross_building(self):
+            self.output.warn("This Corrade recipe could not prepated for cross building")
```
Comment (Croydon) [inline recipes/corrade/all/conanfile.py:51]: `prepated`? Typo?

**Pull request merged**